Swiftfin/Views/LibraryView.swift

```diff
@@ -79,7 +79,23 @@ struct LibraryView: View {
         }
         .toolbar {
             ToolbarItemGroup(placement: .navigationBarTrailing) {
-
+                if viewModel.isLoading && !viewModel.items.isEmpty {
+                    ProgressView()
+                }
+                LibraryViewShuffle()
```

Comment:
Indeed, using a menu makes much more sense, and I renamed to classes to use the term "Random" since it reflects what the feature is more than "Shuffle" does. Also, the LibraryView and BasicLibraryView do seem to be doing really similar things. For now, I added the random button to both, but yeah I do see these two being combined in the future.